### ISSUE TITLE
Code lens does not appear on `.sc` files for `scala` Build server, but do appear for `scala-cli` Build server

### ISSUE THREAD

**Issue by bishabosha** [scalameta/metals#5116]
### Describe the bug

- Go to VS Code with metals plugin
- open a new directory
- create file foo.sc
- in terminal run `scala setup-ide foo.sc` (Scala CLI 1.0.0-RC1)
- in output tab for metals, we see it connects to scala `2023.04.04 13:36:31 INFO  Connected to Build server: scala v1.0.0-RC1`
- **no code lenses appear**.
- now in terminal run `scala-cli setup-ide foo.sc`
- run command Metals: switch build server and choose scala-cli
- in output tab for metals, we see it connects to scala `2023.04.04 13:36:31 INFO  Connected to Build server: scala-cli v1.0.0-RC1`
- **now code lenses appear at the top of the script**

### Expected behavior

I would expect code lenses for either version

### Operating system

macOS

### Editor/Extension

VS Code

### Version of Metals

0.11.11

### Extra context or search terms

scala-cli CLI script file

**Comment by bishabosha** [comment]
I see this as problematic for the workflow where I create a `.sc` file, click import as Scala CLI, then run `scala run foo.sc` from command lines and then quit VS Code. When I re open the file metals will pick up the BSP file for scala and now code lenses disappear

**Comment by tgodzik** [comment]
Thanks for noticing this! I am fixing it in the linked PR. 

> Secondly, I tried in VSCode to override the setting metals.scalaCliLauncher to use scala (for consistency) but it seems the setting is not actually picked up by metals

This is only used if launching ScalaCLI without .bsp folder for a single script file.

**Comment by bishabosha** [comment]
> > Secondly, I tried in VSCode to override the setting metals.scalaCliLauncher to use scala (for consistency) but it seems the setting is not actually picked up by metals
> 
> This is only used if launching ScalaCLI without .bsp folder for a single script file.

That's right, I opened a blank directory, created `foo.sc` and clicked the button when prompted to import as Scala CLI, then I could tell it used the one in my path (because I tried setting `scalaCliLauncher` to my local mill build of `scala-cli`)

**Comment by mindlid** [comment]
confirmed as working current pre-release version, both code lens `run/debug` and docs preview on code hover
